mypyc/irbuild/specialize.py

```diff
@@ -373,9 +374,16 @@ def translate_str_format(
 
         literals = split_braces(format_str)
 
-        variables = [builder.accept(x) if is_str_rprimitive(builder.node_type(x))
-                     else builder.call_c(str_op, [builder.accept(x)], expr.line)
-                     for x in expr.args]
+        # Convert variables to strings
+        variables = []
+        for x in expr.args:
+            if is_str_rprimitive(builder.node_type(x)):
+                var_str = builder.accept(x)
+            elif is_int_rprimitive(builder.node_type(x)):
```

Comment:
Also accept short ints (`is_short_int_rprimitive`). They can use the same code path.